charts/spark-operator-chart/crds/sparkoperator.k8s.io_scheduledsparkapplications.yaml

```diff
@@ -18,4479 +17,11598 @@ spec:
     singular: scheduledsparkapplication
   scope: Namespaced
   versions:
-    - name: v1beta2
-      served: true
-      storage: true
-      subresources:
-        status: {}
-      additionalPrinterColumns:
-        - jsonPath: .spec.schedule
-          name: Schedule
-          type: string
-        - jsonPath: .spec.suspend
-          name: Suspend
-          type: boolean
-        - jsonPath: .status.lastRun
-          name: Last Run
-          type: date
-        - jsonPath: .status.lastRunName
-          name: Last Run Name
-          type: string
-        - jsonPath: .metadata.creationTimestamp
-          name: Age
-          type: date
-      schema:
-        openAPIV3Schema:
-          properties:
-            apiVersion:
-              type: string
-            kind:
-              type: string
-            metadata:
-              type: object
-            spec:
-              properties:
-                concurrencyPolicy:
-                  type: string
-                failedRunHistoryLimit:
-                  format: int32
-                  type: integer
-                schedule:
-                  type: string
-                successfulRunHistoryLimit:
-                  format: int32
-                  type: integer
-                suspend:
-                  type: boolean
-                template:
-                  properties:
-                    arguments:
-                      items:
-                        type: string
-                      type: array
-                    batchScheduler:
+  - additionalPrinterColumns:
```

Comment:
Reading the git diff, it's not clear to me what has actually changed here.  Can you provide a summary of the changes, especially highlighting any breaking changes?

Reply:
The CRDs is generated by `make manifests`, this will generate crd files to `config/crd/bases`, and `make updata-crd` will copy the crd files to `charts/spark-operator-chart/crds/`.

There is no breaking changes to the CRD files, most of the change is  indentation and description. Now you can check out the meaning of CRD fields as follows (before, the description is missed):

```bash
$ kubectl explain sparkapplication.spec.driver
GROUP:      sparkoperator.k8s.io
KIND:       SparkApplication
VERSION:    v1beta2

FIELD: driver <Object>

DESCRIPTION:
    Driver is the driver specification.

FIELDS:
  affinity	<Object>
    Affinity specifies the affinity/anti-affinity settings for the pod.

  annotations	<map[string]string>
    Annotations are the Kubernetes annotations to be added to the pod.

  configMaps	<[]Object>
    ConfigMaps carries information of other ConfigMaps to add to the pod.

  coreLimit	<string>
    CoreLimit specifies a hard limit on CPU cores for the pod.
    Optional

  coreRequest	<string>
    CoreRequest is the physical CPU core request for the driver.
    Maps to `spark.kubernetes.driver.request.cores` that is available since
    Spark 3.0.

  cores	<integer>
    Cores maps to `spark.driver.cores` or `spark.executor.cores` for the driver
    and executors, respectively.

  dnsConfig	<Object>
    DnsConfig dns settings for the pod, following the Kubernetes specifications.

  env	<[]Object>
    Env carries the environment variables to add to the pod.

  envFrom	<[]Object>
    EnvFrom is a list of sources to populate environment variables in the
    container.

  envSecretKeyRefs	<map[string]Object>
    EnvSecretKeyRefs holds a mapping from environment variable names to
    SecretKeyRefs.
    Deprecated. Consider using `env` instead.

  envVars	<map[string]string>
    EnvVars carries the environment variables to add to the pod.
    Deprecated. Consider using `env` instead.

  gpu	<Object>
    GPU specifies GPU requirement for the pod.

  hostAliases	<[]Object>
    HostAliases settings for the pod, following the Kubernetes specifications.

  hostNetwork	<boolean>
    HostNetwork indicates whether to request host networking for the pod or not.

  image	<string>
    Image is the container image to use. Overrides Spec.Image if set.

  initContainers	<[]Object>
    InitContainers is a list of init-containers that run to completion before
    the main Spark container.

  javaOptions	<string>
    JavaOptions is a string of extra JVM options to pass to the driver. For
    instance,
    GC settings or other logging.

  kubernetesMaster	<string>
    KubernetesMaster is the URL of the Kubernetes master used by the driver to
    manage executor pods and
    other Kubernetes resources. Default to https://kubernetes.default.svc.

  labels	<map[string]string>
    Labels are the Kubernetes labels to be added to the pod.

  lifecycle	<Object>
    Lifecycle for running preStop or postStart commands

  memory	<string>
    Memory is the amount of memory to request for the pod.

  memoryOverhead	<string>
    MemoryOverhead is the amount of off-heap memory to allocate in cluster mode,
    in MiB unless otherwise specified.

  nodeSelector	<map[string]string>
    NodeSelector is the Kubernetes node selector to be added to the driver and
    executor pods.
    This field is mutually exclusive with nodeSelector at SparkApplication level
    (which will be deprecated).

  podName	<string>
    PodName is the name of the driver pod that the user creates. This is used
    for the
    in-cluster client mode in which the user creates a client pod where the
    driver of
    the user application runs. It's an error to set this field if Mode is not
    in-cluster-client.

  podSecurityContext	<Object>
    PodSecurityContext specifies the PodSecurityContext to apply.

  ports	<[]Object>
    Ports settings for the pods, following the Kubernetes specifications.

  schedulerName	<string>
    SchedulerName specifies the scheduler that will be used for scheduling

  secrets	<[]Object>
    Secrets carries information of secrets to add to the pod.

  securityContext	<Object>
    SecurityContext specifies the container's SecurityContext to apply.

  serviceAccount	<string>
    ServiceAccount is the name of the custom Kubernetes service account used by
    the pod.

  serviceAnnotations	<map[string]string>
    ServiceAnnotations defines the annotations to be added to the Kubernetes
    headless service used by
    executors to connect to the driver.

  serviceLabels	<map[string]string>
    ServiceLabels defines the labels to be added to the Kubernetes headless
    service used by
    executors to connect to the driver.

  shareProcessNamespace	<boolean>
    ShareProcessNamespace settings for the pod, following the Kubernetes
    specifications.

  sidecars	<[]Object>
    Sidecars is a list of sidecar containers that run along side the main Spark
    container.

  terminationGracePeriodSeconds	<integer>
    Termination grace period seconds for the pod

  tolerations	<[]Object>
    Tolerations specifies the tolerations listed in ".spec.tolerations" to be
    applied to the pod.

  volumeMounts	<[]Object>
    VolumeMounts specifies the volumes listed in ".spec.volumes" to mount into
    the main container's filesystem.
```